### PR TITLE
fix: Adds code to extend DO base class

### PR DESCRIPTION
**Pull request by megaconfidence**
### Summary

This PR addx the `className extends DurableObject` to code sinppets to clearly show that DO classes must extend the base class, thus avoiding reader confusion

### CODE REVIEW

File: src/content/docs/durable-objects/api/alarms.mdx [hunk 1]
```diff
@@ -120,4 +122,4 @@ export class AlarmExample {
 
 * Understand how to [use the Alarms API](/durable-objects/examples/alarms-api/) in an end-to-end example.
 * Read the [Durable Objects alarms announcement blog post](https://blog.cloudflare.com/durable-objects-alarms/).
-* Review the [Transactional Storage API](/durable-objects/api/transactional-storage-api/) documentation for Durable Objects.
+* Review the [Transactional Storage API](/durable-objects/api/transactional-storage-api/) documentation for Durable Objects.
```
Comment (vy-ton) [inline src/content/docs/durable-objects/api/alarms.mdx]: Can you adjust PR to not have these changes?

**Comment by joshthoward** [comment]
@megaconfidence you do not need to explicitly have the Durable Object extend from the base class unless you are using RPC. I don't think that this is bad to show since customers should be using RPC when possible.

File: src/content/docs/durable-objects/api/alarms.mdx (hunk 1 above)
Comment (megaconfidence) [reply src/content/docs/durable-objects/api/alarms.mdx]: Sorry for these little annoying changes, but I don't think these files have been formatted as per the new editorconfig being used with the switch to Astro https://github.com/cloudflare/cloudflare-docs/blob/f51bf3f91cb62fec026e2bb2baf6944feb2f80c1/.editorconfig#L9

File: src/content/docs/durable-objects/api/transactional-storage-api.mdx [hunk 2]
```diff
@@ -15,7 +15,7 @@ While access to a Durable Object instance is single-threaded, request executions
 The following code snippet shows you how to store and retrieve data using the Transactional Storage API.
 
 ```js
-export class Counter {
+export class Counter extends DurableObject {
```
Comment (vy-ton) [inline src/content/docs/durable-objects/api/transactional-storage-api.mdx]: needs import

File: src/content/docs/durable-objects/get-started.mdx [hunk 3]
```diff
@@ -69,15 +69,15 @@ Your `MyDurableObject` class will have a constructor with two parameters. The fi
 <Tabs> <TabItem label="JavaScript" icon="seti:javascript">
 
 ```js
-export class MyDurableObject {
+export class MyDurableObject extends DurableObject {
```
Comment (vy-ton) [inline src/content/docs/durable-objects/get-started.mdx]: needs import

File: src/content/docs/durable-objects/api/transactional-storage-api.mdx (hunk 2 above)
Comment (megaconfidence) [reply src/content/docs/durable-objects/api/transactional-storage-api.mdx]: Not really, this piece of code only shows a snippet and not the full DO. Adding an import might be distracting/verbose

File: src/content/docs/durable-objects/get-started.mdx (hunk 3 above)
Comment (megaconfidence) [reply src/content/docs/durable-objects/get-started.mdx]: Not really, this piece of code only shows a snippet and not the full DO. Adding an import might be distracting/verbose

**Review by vy-ton — APPROVED** [review]

**Comment by megaconfidence** [comment]
Hi @joshthoward, can you also approve this PR? Thanks

**Review by KimJ15 — APPROVED** [review]
Approved from a prose perspective on behalf of Kody

**Comment by megaconfidence** [comment]
Hey @KianNH, can you approve?

**Pull request merged**